packages/x-data-grid-premium/src/hooks/utils/useGridAriaAttributes.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import {
+  useGridAriaAttributes as useGridAriaAttributesPro,
+  useGridSelector,
+} from '@mui/x-data-grid-pro/internals';
+import { useGridPrivateApiContext } from './useGridPrivateApiContext';
+import { gridRowGroupingSanitizedModelSelector } from '../features/rowGrouping/gridRowGroupingSelector';
+
+export const useGridAriaAttributes = (): React.HTMLAttributes<HTMLElement> => {
+  const ariaAttributesPro = useGridAriaAttributesPro();
+  const apiRef = useGridPrivateApiContext();
+  const gridRowGroupingModel = useGridSelector(apiRef, gridRowGroupingSanitizedModelSelector);
+
+  const ariaAttributesPremium = gridRowGroupingModel.length > 0 ? { role: 'treegrid' } : {};
```

Comment:
When I changed the role from "grid" to "treegrid" for the tree data feature in https://github.com/mui/mui-x/pull/9496, I added the `ariaV7` experimental flag to avoid breaking changes.
While it's technically a "bugfix", perhaps we should do something similar here?
What do you think?

Reply:
I agree. Thanks for the reference
I have added a flag for the premium grid

Reply:
We also need to document it, similar to https://v6.mui.com/x/react-data-grid/accessibility/#accessibility-changes-in-v7
And how about enabling it in the demos that are using `useDemoData`? https://github.com/mui/mui-x/pull/10204

Reply:
I have added the docs under the row grouping page, since this flag only affects grid with that feature.
I have also created a cleanup ticket and linked in in the umbrella ticket for the v8 preparation

Only full example grid is actually using demo data, the rest of them are using movie data hook for getting the content. I didn't want to pollute that hook since it is used everywhere, so I have explicitly added a new example in the accessibility section and enabled the experimental flag in the full example